FanGraphSpider.py

```diff
@@ -1,174 +0,0 @@
-                        name = namespot.xpath('a/text()').extract_first()
-                        if plname == name:
-                            url = namespot.xpath('a/@href').extract_first()
-                            break
-                    url = response.urljoin(url)
-                    yield SplashRequest(url, callback=self.steal_parse, args=dict(self.SPLASH_ARGS))
-
-    # Gathers player stats and constructs tweets for homeruns
-    def home_parse(self, response):
-        p_name = response.xpath('//div[@class="player-info-box-name"]/h1/text()').extract_first()
-        team = response.xpath('//div[@class="player-info-box-name-team"]/a/text()').extract_first()
-        av = response.xpath('//table[@class="rgMasterTable"][1]/tbody/tr[2]/td[17]/text()').extract_first()
-        rstat = response.xpath('//table[@class="rgMasterTable"][1]/tbody/tr[2]/td[7]/text()').extract_first()
-        hr = response.xpath('//table[@class="rgMasterTable"][1]/tbody/tr[2]/td[6]/text()').extract_first()
-        rbi = response.xpath('//table[@class="rgMasterTable"][1]/tbody/tr[2]/td[8]/text()').extract_first()
-        sb = response.xpath('//table[@class="rgMasterTable"][1]/tbody/tr[2]/td[9]/text()').extract_first()
-        obp = response.xpath('//table[@class="rgMasterTable"][1]/tbody/tr[2]/td[18]/text()').extract_first()
-        slg = response.xpath('//table[@class="rgMasterTable"][1]/tbody/tr[2]/td[19]/text()').extract_first()
-        OPS = str(float(obp)+float(slg))
-
-        tweet = u'\U0001F929' + ' Home Run! ' + u'\U0001F929' + '- %s - %s \n Season Totals \n %s AVG, %s R, %s HR, %s RBI, %s SB, %s OBP, %s OPS \n #homerun #fantasybaseball' % (p_name, team, av, rstat, hr, rbi, sb, obp, OPS)
-
-
-        auth = tweepy.OAuthHandler(tkey, tsecret)
-        auth.set_access_token(atoke, asecret)
-
-        api = tweepy.API(auth)
-
-        api.update_status(tweet)
-
-
-    # Gathers player stats and constructs tweets for Stolen Bases
-    def steal_parse(self, response):
-
-        p_name = response.xpath('//div[@class="player-info-box-name"]/h1/text()').extract_first()
-        team = response.xpath('//div[@class="player-info-box-name-team"]/a/text()').extract_first()
-        av = response.xpath('//table[@class="rgMasterTable"][1]/tbody/tr[2]/td[17]/text()').extract_first()
-        rstat = response.xpath('//table[@class="rgMasterTable"][1]/tbody/tr[2]/td[7]/text()').extract_first()
-        hr = response.xpath('//table[@class="rgMasterTable"][1]/tbody/tr[2]/td[6]/text()').extract_first()
-        rbi = response.xpath('//table[@class="rgMasterTable"][1]/tbody/tr[2]/td[8]/text()').extract_first()
-        sb = response.xpath('//table[@class="rgMasterTable"][1]/tbody/tr[2]/td[9]/text()').extract_first()
-        obp = response.xpath('//table[@class="rgMasterTable"][1]/tbody/tr[2]/td[18]/text()').extract_first()
-        slg = response.xpath('//table[@class="rgMasterTable"][1]/tbody/tr[2]/td[19]/text()').extract_first()

```

Comment:
All of these xpath strings could be pulled up to the top of the file in a configuration dictionary as well, like
```
xpath_map = {
    'p_name' : '//div[@class="player-info-box-name-team"]/a/text()',
    ...
}
```
which is easier to see at the top instead of down in a method, and easier to change. The next step past that (for a "real world" program) would be building that dictionary from an outside config file, but eh.